packages/hashtower/src/add.ts

```diff
@@ -0,0 +1,34 @@
+import checkParameter from "./checkParameter"
+import { HashFunction } from "./types"
+
+export default function _add(
```

Comment:
I'm _really_ not a fan of this separate file, separate function design. It's super hard to follow what is happening because we're passing properties from the main instance by reference and then mutating them.

I'd much rather see 1 file that is 200 lines than 5 files each < 50 lines.

Reply:
Re 1 file with 200 lines vs 4 with 50 lines, I agree in general, but I remember there were some differences in terms of performance. I need to check it again

Reply:
I think most of the computation is on poseidon(), the data structure itself consume very little time.

However, the time of poseidon() itself varies about 10% on my computer between different runs.

```javascript
import { poseidon } from "circomlibjs";

console.time("1")
for (let i = 0; i < 10000; i++) { poseidon([i, i + 1]) }
console.timeEnd("1")

console.time("1")
for (let i = 0; i < 10000; i++) { poseidon([i, i + 1]) }
console.timeEnd("1")
```

```shell
hashtower % node a.mjs
1: 5.573s
1: 5.611s
hashtower % node a.mjs
1: 5.751s
1: 5.820s
hashtower % node a.mjs
1: 5.735s
1: 5.969s
hashtower % node a.mjs
1: 5.862s
1: 5.739s
hashtower % node a.mjs
1: 6.054s
1: 5.953s
```

Reply:
You should try benchmarking `poseidon-lite`, it might be faster. In the latest version i removed some of the modulo reduction to trade memory for compute. I was seeing < 1 ms / hash even using implementation without pre-multiplied constants.

Reply:
`poseidon-lite` is about 50% faster.
https://gist.github.com/LCamel/f9f1c377af8dd6aba48f2c68ae9ce80e

Reply:
<img src="https://user-images.githubusercontent.com/631020/233667261-632daac4-d240-450c-b4b9-2721dc2e909c.png" width="30" />